src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,189 @@
+        } else {
+            Task task = list.get(index - 1);
+            task.markAsDone();
+            String result = "Nice! I have marked this task as done!\n" + task.toString() ;
+            return result;
+        }
+    }
+
+    /**
+     * A method to add a Task inside a TaskList
+     *
+     * @param str User inputs which indicates details of the task to be added.
+     * @return Task after it is added.
+     * @throws EmptyDescriptionException Thrown when users do not give a task description.
+     * @throws InvalidTaskException Thrown when users give incorrect input for task.
+     * @throws InvalidDeadlineException Thrown when users give an incorrect deadline.
+     */
+
+    public String addTask(String str) throws EmptyDescriptionException
+            , InvalidTaskException, InvalidDeadlineException {
+        //first check if the task only contain 1 word
+        if (str.split(" ").length == 1) {
+            //check if task if valid
+            if (str.startsWith("todo") || str.startsWith("deadline") || str.startsWith("event")) {
+                throw new EmptyDescriptionException(str);
+            } else { //check for invalid task
+                throw new InvalidTaskException();
+            }
+        }
+
+        //task that contain more than 1 word.
+        //check for invalid task. otherwise, add the correct task to the list.
+        else {
+            if (!str.startsWith("todo") && !str.startsWith("deadline") && !str.startsWith("event")) {
+                throw new InvalidTaskException();
+            }
+            else {
+                Task task;
+                if (str.startsWith("todo")) {
+                    task = new ToDos(str.substring(5));
+                    list.add(task);
+                } else if (str.startsWith("deadline")) {
+                    String[] message = str.split("/by ");
+                    if (message.length == 1) {
+                        throw new InvalidTaskException();
+                    } else {
+                        task = new Deadline(message[0].substring(9), message[1]);
+                        list.add(task);
+                    }
+                } else {
+                    String[] message = str.split("/at ");
+                    if (message.length == 1) {
+                        throw new InvalidTaskException();
+                    } else {
+                        task = new Events(message[0].substring(6), message[1]);
+                        list.add(task);
+                    }
+                }
+
+                String result = "Got it. I've added this task.\n"
+                        + list.get(list.size() - 1).toString() + "\r\n"
+                        + "Now you have " + list.size()
+                        + (list.size() == 1 ? " task in the list" : " tasks in the list.");
+                return result;
+            }
+        }
+    }
+
+    /**
+     * A method to delete a certain in a TaskList.
```

Comment:
Perhaps an empty line between the parameter/return/throws section and the description?